lib/bulk_processor/csv_processor/row_processor.rb

```diff
@@ -0,0 +1,61 @@
+class BulkProcessor
+  class CSVProcessor
+    # An abstract implementation of the RowProcessor role. This class implements
+    # `#results` by returning an array of `Results`. To subclass, just implement
+    # `#process` to handle the row.
+    #
+    # The row will be considered a failure by default. After a row is successfully
+    # processed, set `self.successful = true`. If there are any messages that
+    # should be passed back to the Handler, add them to the `#errors` array.
```

Comment:
What is `successful?` were based on `errors.empty?`? Then we could enforce that failures include a useful reason.